source/game/field/BoxColManager.cc

```diff
@@ -0,0 +1,620 @@
+#include "BoxColManager.hh"
+
+#include <numeric>
+
+namespace Field {
+
+/// @addr{0x80786ED0}
+BoxColUnit::BoxColUnit() {
+    m_pos = nullptr;
+    m_radius = 0.0f;
+    m_range = 0.0f;
+    m_userData = nullptr;
+}
+
+/// @addr{0x80786EF4}
+BoxColUnit::~BoxColUnit() = default;
+
+/// @addr{0x80786F34}
+void BoxColUnit::init(f32 radius, f32 maxSpeed, const EGG::Vector3f *pos, BoxColFlag flag,
+        void *userData) {
+    m_pos = pos;
+    m_radius = radius;
+    m_range = radius + maxSpeed;
+    m_flag = flag;
+    m_userData = userData;
+    m_xMax = pos->x + m_range;
+    m_xMin = pos->x - m_range;
+}
+
+/// @addr{0x80786F6C}
+void BoxColUnit::makeDynamic() {
+    m_flag.resetBit(eBoxColFlag::STATIC);
+}
+
+/// @addr{0x80786F7C}
+void BoxColUnit::resize(f32 radius, f32 maxSpeed) {
+    m_radius = radius;
+    m_range = radius + maxSpeed;
+    m_flag.setBit(eBoxColFlag::DYNAMIC);
+}
+
+/// @addr{0x80786F98}
+void BoxColUnit::reinsert() {
+    BoxColManager::Instance()->reinsertUnit(this);
+}
+
+/// @addr{0x80786FA8}
+void BoxColUnit::search(BoxColFlag flag) {
+    BoxColManager::Instance()->search(this, flag);
+}
+
+/// @addr{0x807856E0}
+BoxColManager::BoxColManager() {
+    constexpr float UPPER_BOUND_VAL = 999999.9f;
+
+    static EGG::Vector3f upperBound(UPPER_BOUND_VAL, UPPER_BOUND_VAL, UPPER_BOUND_VAL);
+    static EGG::Vector3f lowerBound(-UPPER_BOUND_VAL, -UPPER_BOUND_VAL, -UPPER_BOUND_VAL);
+
+    std::iota(m_unitIDs.begin(), m_unitIDs.end(), 1);
+
+    m_unitCount = 0;
+    m_nextUnitID = 0;
+
+    clear();
+
+    BoxColFlag flags;
+    insert(1.0f, 0.0f, &upperBound, flags, nullptr)->m_flag.setBit(eBoxColFlag::DISABLED);
+    insert(1.0f, 0.0f, &lowerBound, flags, nullptr)->m_flag.setBit(eBoxColFlag::DISABLED);
+}
+
+/// @addr{0x807854e4}
+BoxColManager::~BoxColManager() = default;
+
+/// @addr{0x8078597C}
+void BoxColManager::clear() {
+    m_nextPlayerID = MAX_UNIT_COUNT;
+    m_nextItemID = MAX_UNIT_COUNT;
+    m_nextObjectID = MAX_UNIT_COUNT;
+    m_nextDrivableID = MAX_UNIT_COUNT;
+    m_maxID = 0;
+    m_cacheQueryUnit = nullptr;
+    m_cacheRadius = -1.0f;
+    m_cacheFlag.makeAllZero();
+}
+
+/// @addr{0x807859B0}
+void BoxColManager::calc() {
```

Comment:
Rough outline of the Doxygen brief: we need to recalculate the boundaries of all active units with a RecalcAABB flag, and then update the low and high points accordingly.